test/unittests/OpModel/TTNN/Lib/TestOpModelLib.cpp

```diff
@@ -0,0 +1,660 @@
+  EXPECT_EQ(cb_size, 8192);
+  EXPECT_EQ(output_size, 0);
+  EXPECT_EQ(peak_size, 0);
+
+  opConstraint = ReluOpInterface::getOpConstraints(
+      tensorShape, inputLayout_dram, tensorShape, inputLayout_l1, workerGrid);
+  std::tie(legal, l1Usage, errorMsg) = opConstraint;
+  EXPECT_TRUE(legal);
+  EXPECT_TRUE(l1Usage.has_value());
+  EXPECT_FALSE(errorMsg.has_value());
+  std::tie(cb_size, peak_size, output_size) = l1Usage.value();
+  EXPECT_EQ(cb_size, 8192);
+  EXPECT_EQ(output_size, 4096);
+  EXPECT_EQ(peak_size, 4096);
+
+  opConstraint = ReluOpInterface::getOpConstraints(
+      tensorShape, inputLayout_l1, tensorShape, inputLayout_dram, workerGrid);
+  std::tie(legal, l1Usage, errorMsg) = opConstraint;
+  EXPECT_TRUE(legal);
+  EXPECT_TRUE(l1Usage.has_value());
+  EXPECT_FALSE(errorMsg.has_value());
+  std::tie(cb_size, peak_size, output_size) = l1Usage.value();
+  EXPECT_EQ(cb_size, 8192);
+  EXPECT_EQ(output_size, 0);
+  EXPECT_EQ(peak_size, 0);
+
+  opConstraint = ReluOpInterface::getOpConstraints(
+      tensorShape, inputLayout_l1, tensorShape, inputLayout_l1, workerGrid);
+  std::tie(legal, l1Usage, errorMsg) = opConstraint;
+  EXPECT_TRUE(legal);
+  EXPECT_TRUE(l1Usage.has_value());
+  EXPECT_FALSE(errorMsg.has_value());
+  std::tie(cb_size, peak_size, output_size) = l1Usage.value();
+  EXPECT_EQ(cb_size, 8192);
+  EXPECT_EQ(output_size, 4096);
+  EXPECT_EQ(peak_size, 4096);
+}
+
+TEST_F(OpModelBase, ReluSharded) {
+  const llvm::SmallVector<int64_t> tensorShape = {16 * 64 * 32, 32};
+  const auto workerGrid = CreateWorkerGrid(
+      mlir::tt::ttnn::TensorMemoryLayout::HeightSharded, {8, 8});
+  const auto virtualGrid = GetVirtualGridShape(
+      tensorShape, mlir::tt::ttnn::TensorMemoryLayout::HeightSharded, {8, 8});
+  const mlir::tt::ttnn::TTNNLayoutAttr inputLayout_l1_hs = CreateLayout(
+      tensorShape, mlir::tt::ttnn::BufferType::L1,
+      mlir::tt::ttnn::TensorMemoryLayout::HeightSharded, virtualGrid);
+  const mlir::tt::ttnn::TTNNLayoutAttr inputLayout_l1_i =
+      CreateLayout(tensorShape, mlir::tt::ttnn::BufferType::L1,
+                   mlir::tt::ttnn::TensorMemoryLayout::Interleaved);
+
+  std::tuple<bool, std::optional<std::tuple<size_t, size_t, size_t>>,
+             std::optional<std::string>>
+      opConstraint;
+  bool legal = false;
+  std::optional<std::tuple<size_t, size_t, size_t>> l1Usage = std::nullopt;
+  std::optional<std::string> errorMsg = "";
+  size_t cb_size = 0;
+  size_t peak_size = 0;
+  size_t output_size = 0;
+
+  opConstraint = ReluOpInterface::getOpConstraints(
+      tensorShape, inputLayout_l1_hs, tensorShape, inputLayout_l1_hs,
+      workerGrid);
+  std::tie(legal, l1Usage, errorMsg) = opConstraint;
+  EXPECT_TRUE(legal);
+  EXPECT_TRUE(l1Usage.has_value());
+  EXPECT_FALSE(errorMsg.has_value());
+  std::tie(cb_size, peak_size, output_size) = l1Usage.value();
+  EXPECT_EQ(cb_size, 0);
+  EXPECT_EQ(output_size, tensorShape[0] * tensorShape[1] * 2 / 64);
```

Comment:
that's not entirely and always true; 

there are two folds to this observation:
- grid phy cores is really 7x8 on this N300, not 64. I need to add a check of provided workerGrid versus a real device.
- never the less, as interleaved allocation always starts with bank 0, it can take up to num_banks -1 more memory than equivalent sharded tensor; 

in this specific case this appeared as x2.